kamon-play-2.6.x/src/main/scala/kamon/play/instrumentation/RequestHandlerInstrumentation.scala

```diff
@@ -15,48 +15,54 @@
 
 package kamon.play.instrumentation
 
-import akka.http.scaladsl.model.{HttpRequest, HttpResponse}
 import kamon.Kamon
 import kamon.context.Context
-import kamon.play.OperationNameFilter
 import kamon.trace.Span
 import kamon.util.CallingThreadExecutionContext
-import org.aspectj.lang.ProceedingJoinPoint
-import org.aspectj.lang.annotation._
-import play.api.mvc.EssentialFilter
 
 import scala.concurrent.Future
 
-@Aspect
-class RequestHandlerInstrumentation {
+trait GenericRequest {
+  val headers: Map[String, String]
+  val method: String
+  val url: String
+  val spanKind: String
```

Comment:
The `spanKind` name is misleading because it is actually being used for the `component` tag, it should be changed.